appinventor/components/src/com/google/appinventor/components/runtime/ReplForm.java

```diff
@@ -124,20 +124,73 @@ public void onCreate(Bundle icicle) {
   }
 
   @Override
   void onCreateFinish() {
     super.onCreateFinish();
+    Log.d(LOG_TAG, "onCreateFinish() Called in Repl");
 
     if (!isEmulator() && AppInventorFeatures.doCompanionSplashScreen())
       {                    // Only show REPL splash if not in emulator and enabled
         Intent webviewIntent = new Intent(Intent.ACTION_MAIN);
         webviewIntent.setClassName(activeForm.$context(), SPLASH_ACTIVITY_CLASS);
         activeForm.$context().startActivity(webviewIntent);
       }
+    Intent intent = getIntent();
+    Log.d(LOG_TAG, "Intent = " + intent);
+    final String data = intent.getDataString();
+    if (data != null) {
+      Log.d(LOG_TAG, "Got data = " + data);
+    } else {
+      Log.d(LOG_TAG, "Did not receive any data");
+    }
+
+/////////////////////////////////////////////////////////////////////////
+// Chromebook Support:                                                 //
+//                                                                     //
+// The code below parses the data provided in the intent to get the    //
+// code to use to talk to the rendezvous server. It appears to need to //
+// run on the UI thread (I'm not sure what thread we are on here in    //
+// onCreateFinish().  I'm not sure why we need the delay, but it       //
+// doesn't work if we do not include the delay. I'm continuing to look //
+// into why that is and the delay may be removed in a future           //
+// revision. (jis).                                                    //
+//                                                                     //
+// Also: the rendezvous server location is hardcoded in this version.  //
+// a future version will let you customize the location of the         //
+// rendezvous server.                                                  //
+/////////////////////////////////////////////////////////////////////////
+
+    if (data != null && (data.startsWith("aicompanion"))) {
+      androidUIHandler.postDelayed(new Runnable() {
+          @Override
+          public void run() {
+            String code = data.substring(data.indexOf("//comp/") + 7);
+            PhoneStatus status = new PhoneStatus(ReplForm.this);
+            status.WebRTC(true);
+            code = status.setHmacSeedReturnCode(code, "rendezvous.appinventor.mit.edu");
+            String ipAddress = PhoneStatus.GetWifiIpAddress();
+            int api = status.SdkLevel();
+            String version = status.GetVersionName();
+            String aid = status.InstallationId();
+            Log.d(LOG_TAG, "InstallationId = " + aid);
+            boolean webrtc = true;
+            Web web = new Web(ReplForm.this);
+            web.Url("http://rendezvous.appinventor.mit.edu/rendezvous/");
+            web.PostText("ipaddr=" + ipAddress + "&port=9987&webrtc=" + webrtc
+              + "&version=" + version + "&api=" + api + "&aid=" +
+              aid + "&installer=unknown&r2=true&key=" + code);
```

Comment:
Any reason to not use `status.GetInstaller()`?

Reply:
Nope, now I do :-)